File: frontend/features/get_messages.js

async function getmessages(conversation_id,token) {
    try {
    const data = await fetch(`http://localhost:8000/chat/get-messages`,
        {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Authorization": `Bearer ${token}`
            },
            body: JSON.stringify({
                "conversation_id": conversation_id
            })

        }
    )
    const response = await data
    return response.json()
    }
    catch (error) {
        console.log(error)
        return {"error": `Error fetching messages ${error.message}`}
    }  
}

export default getmessages

File: backend/services/agent/utils/getMessages.py
from urllib import response

from fastapi.encoders import jsonable_encoder
from fastapi.responses import JSONResponse
import httpx
import os


async def get_Messages(conversation_id):
    try:
        async with httpx.AsyncClient() as client:
            response = await client.post(
                os.getenv("CHAT_SERVICE_URL") + "/get-messages",
                headers={
                    "Authorization": os.getenv("CHAT_SERVICE_AUTHORIZATION")
                },
                json={
                    "conversation_id": conversation_id
                }
            )
        data = response.json()
        return data.get("messages", data if isinstance(data, list) else [])
    except Exception as e:
        return JSONResponse(
            status_code=400,
            content=jsonable_encoder({"get messages error": str(e)})
        )